akka-stream/src/main/resources/reference.conf

```diff
@@ -45,6 +45,16 @@ akka {
       # this may cause an initial runtime overhead, but most of the time fusing is
       # desirable since it reduces the number of Actors that are created.
       auto-fusing = on
+      
+      # Those stream elements which have explicit buffers (like mapAsync, mapAsyncUnordered,
+      # buffer, flatMapMerge, Source.actorRef, Source.queue, etc.) will preallocate a fixed
+      # buffer upon stream materialization if the requested buffer size is less than this
+      # configuration parameter. The default is very high because failing early is better
+      # than failing under load.
+      #
+      # Buffers larger than this value will use a dynamic approach that has a higher
+      # per-element cost.
+      max-fixed-buffer-size = 1000000000
```

Comment:
There would only be an effect if the default is changed to a low value and then some buffers are Bounded while others are Fixed. And BoundedBuffer already becomes bi-morphic if its fixed part overflows and it enters dynamic mode.